express-api/src/services/notifications/notificationServices.ts

```diff
@@ -0,0 +1,206 @@
+  template: NotificationTemplate,
+  projStatusNotif: ProjectStatusNotification,
+  project: Project,
+  agency?: Agency,
+  overrideTo?: string,
+) => {
+  const sendDate = new Date();
+  sendDate.setDate(sendDate.getDate() + projStatusNotif.DelayDays);
+  const queueObject = {
+    Key: randomUUID(),
+    Status: NotificationStatus.Pending,
+    Priority: template.Priority,
+    Encoding: template.Encoding,
+    SendOn: sendDate,
+    Subject: nunjucks.renderString(template.Subject, { Project: project }),
+    Body: nunjucks.renderString(template.Body, {
+      Title: Title,
+      Uri: Uri,
+      ToAgency: agency,
+      Project: flattenProjectProperties(project),
+    }),
+    TemplateId: template.Id,
+    To: [overrideTo ?? agency?.Email, template.To].filter((a) => a).join(';'),
+    Cc: [agency?.CCEmail, template.Cc].filter((a) => a).join(';'),
+    Bcc: template.Bcc,
+  };
+  const insertedNotif = await AppDataSource.getRepository(NotificationQueue).save(queueObject);
+  return insertedNotif;
+};
+
+const generateProjectNotifications = async (project: Project, previousStatusId: number) => {
+  const projectStatusNotif1 = await AppDataSource.getRepository(ProjectStatusNotification).find({
+    where: { FromStatusId: previousStatusId, ToStatusId: project.StatusId },
+  });
+  const projectStatusNotif2 = await AppDataSource.getRepository(ProjectStatusNotification).find({
+    where: { FromStatusId: IsNull(), ToStatusId: project.StatusId },
+  });
+  const projectStatusNotifications = [...projectStatusNotif1, ...projectStatusNotif2];
+  const returnNotifications = [];
+  for (const projStatusNotif of projectStatusNotifications) {
+    const template = await AppDataSource.getRepository(NotificationTemplate).findOne({
+      where: { Id: projStatusNotif.Id },
+    });
+
+    const sendDate = new Date();
+    sendDate.setDate(sendDate.getDate() + projStatusNotif.DelayDays);
+
+    let overrideTo: string | null = null;
+    if (template.Audience == NotificationAudience.ProjectOwner) {
+      const owningUser = await AppDataSource.getRepository(User).findOne({
+        where: { Id: project.CreatedById },
+      });
+      overrideTo = owningUser.Email;
+      returnNotifications.push(
+        insertProjectNotificationQueue(
+          template,
+          projStatusNotif,
+          project,
+          project.Agency,
+          overrideTo,
+        ),
+      );
+    } else if (template.Audience == NotificationAudience.OwningAgency) {
+      returnNotifications.push(
+        insertProjectNotificationQueue(template, projStatusNotif, project, project.Agency),
+      );
+    } else if (template.Audience == NotificationAudience.Agencies) {
+      //Dunno if we really have/plan to have the ProjectAgencyResponses implemented?
+    } else if (template.Audience == NotificationAudience.ParentAgencies) {
+      //Same here
+    } else if (template.Audience == NotificationAudience.WatchingAgencies) {
+      //Same here
```

Comment:
We can probably fill these in later, but I think there might be a need to notify agencies on a broad basis like this when properties enter the ERP.